usersimcrs/simulator/agenda_based/interaction_model.py

```diff
@@ -488,8 +568,99 @@ def _sample_next_user_dialogue_acts(
         Returns:
             List of dialogue acts.
         """
-        # Check if the agent's dialogue acts are in the compound transition
-        # matrix. If not, we consider the single transition matrix.
+        sampled_user_intents = self._sample_user_intents(agent_dialogue_acts)
+
+        # Generate dialogue acts based on the sampled intents, the annotations
+        # are generated based on the information need and belief state. Note
+        # that only simple cases (disclose and inquire) are considered here.
+        user_dialogue_acts = []
+        current_belief_state = (
+            self.dialogue_state_tracker.get_current_state().belief_state
+        )
+        for sampled_intent in sampled_user_intents:
+            if sampled_intent == self.INTENT_DISCLOSE:  # type: ignore[attr-defined] # noqa
+                # Check if there is a slot from the information need that is
+                # not fulfilled in the belief state, else choose a random slot
+                # value from the constraints.
+                slot = None
+                value = None
+                for belief_state_slot in current_belief_state.keys():
+                    if belief_state_slot not in information_need.constraints:
+                        slot = belief_state_slot
+                        value = information_need.get_constraint_value(slot)
+                        break
+
+                if slot is None:
+                    slot, value = random.choice(
+                        list(information_need.constraints.items())
+                    )
 
-        # TODO: Implement this method
-        pass
+                user_dialogue_acts.append(
+                    DialogueAct(
+                        sampled_intent, [SlotValueAnnotation(slot, value)]
+                    )
+                )
+            elif sampled_intent == self.INTENT_INQUIRE:  # type: ignore[attr-defined] # noqa
+                slot = random.choice(information_need.get_requestable_slots())
+                if not slot:
+                    slot = random.choice(self._domain.get_requestable_slots())
+                user_dialogue_acts.append(
+                    DialogueAct(sampled_intent, [SlotValueAnnotation(slot)])
+                )
+            else:
+                user_dialogue_acts.append(DialogueAct(sampled_intent))
+
+        return user_dialogue_acts
+
+    def _sample_user_intents(self, agent_dialogue_acts: List[DialogueAct]):
+        """Samples user intents based on the agent's dialogue acts.
+
+        Checks if the agent's dialogue acts are in the compound transition
+        matrix. If not, we consider the single transition matrix.
+
+        Args:
+            agent_dialogue_acts: Agent's dialogue acts.
+
+        Returns:
+            List of sampled user intents.
+        """
+        compound_agent_intent = "_".join(
+            sorted({da.intent.label for da in agent_dialogue_acts})
+        )
```

Comment:
Extract this as a method `_get_compound_intent_label(dialogue_acts)`? Could be also used in lines 142-144 and 145-147. The comment I left above on the sorting could then be in the docstring of this method.

Reply:
Good point. Done.